lib/charms/hydra/v0/oauth.py

```diff
@@ -662,6 +694,17 @@ def _delete_juju_secret(self, relation: Relation) -> None:
         secret = self.model.get_secret(label=self._get_secret_label(relation))
         secret.remove_all_revisions()
 
+    def _pop_client_data(self, relation_id: Relation) -> None:
+        if len(self.model.relations) == 0:
+            return None
+
+        relation = self.model.get_relation(self._relation_name, relation_id=relation_id)
+        if not relation or not relation.app:
+            return None
+
+        for provider_data in list(relation.data[self.model.app]):
+            relation.data[self.model.app].pop(provider_data, "")
```

Comment:
I believe that this can fail. The function body is not an atomic operation which means that the data may be gone when this runs (maybe the data was never there). I think we need to wrap this in a try/catch.

Reply:
The relation data is still present on relation -departed and -broken. If it was gone (like you'd expect from the relation docs), this method would not be needed at all. If `relation.data[self.model.app]` was empty for whichever reason, this method would just have nothing to pop.
That said, I think it doesn't hurt to wrap it in a try/catch, so I'll update the PR.